tests/test_validity.py

```diff
@@ -65,9 +65,29 @@ def test_products(stac_fixture, product_id, product_schema):
         [("http://example.com/schema.json", Resource.from_contents(schema))],
     )
     validator = jsonschema.Draft7Validator({"$ref": "http://example.com/schema.json"}, registry=registry)
-    for stac_dict in stac_fixture:
+    for _, stac_dict in stac_fixture:
         is_collection = stac_dict["type"] == "Collection" and stac_dict["id"] == product_id
         is_item = stac_dict["type"] == "Feature" and stac_dict["collection"] == product_id
         if is_collection or is_item:
             error = best_match(validator.iter_errors(stac_dict))
             assert error is None, f"{stac_dict['id']} is invalid. Reason: {error.message}"
+
+
+def test_stac_id(stac_fixture):
+    for stac_filename, stac_dict in stac_fixture:
```

Comment:
I would add some docstrings to the tests with explanation of what they are doing. But in general it seems ok.